src/platform/cc13xx_26xx/FactoryDataProvider.cpp

```diff
@@ -0,0 +1,333 @@
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+CHIP_ERROR FactoryDataProvider::GetSpake2pIterationCount(uint32_t & iterationCount)
+{
+    ReturnErrorCodeIf(!mFactoryData.spake2p_it.data, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+    memset(&iterationCount, 0, 4);
+    memcpy(&iterationCount, mFactoryData.spake2p_it.data, mFactoryData.spake2p_it.len);
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR FactoryDataProvider::GetSpake2pSalt(MutableByteSpan & saltBuf)
+{
+    ReturnErrorCodeIf(saltBuf.size() < mFactoryData.spake2p_salt.len, CHIP_ERROR_BUFFER_TOO_SMALL);
+    ReturnErrorCodeIf(!mFactoryData.spake2p_salt.data, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+    memcpy(saltBuf.data(), mFactoryData.spake2p_salt.data, mFactoryData.spake2p_salt.len);
+    saltBuf.reduce_size(mFactoryData.spake2p_salt.len);
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR FactoryDataProvider::GetSpake2pVerifier(MutableByteSpan & verifierBuf, size_t & verifierLen)
+{
+    ReturnErrorCodeIf(verifierBuf.size() < mFactoryData.spake2p_verifier.len, CHIP_ERROR_BUFFER_TOO_SMALL);
+    ReturnErrorCodeIf(!mFactoryData.spake2p_verifier.data, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+    memcpy(verifierBuf.data(), mFactoryData.spake2p_verifier.data, mFactoryData.spake2p_verifier.len);
+    verifierLen = mFactoryData.spake2p_verifier.len;
+    verifierBuf.reduce_size(verifierLen);
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR FactoryDataProvider::GetSetupPasscode(uint32_t & setupPasscode)
+{
+    ReturnErrorCodeIf(!mFactoryData.passcode.data, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+    memcpy(&setupPasscode, mFactoryData.passcode.data, mFactoryData.passcode.len);
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR FactoryDataProvider::SetSetupPasscode(uint32_t setupPasscode)
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+CHIP_ERROR FactoryDataProvider::GetVendorName(char * buf, size_t bufSize)
+{
+    ReturnErrorCodeIf(bufSize < mFactoryData.vendor_name.len, CHIP_ERROR_BUFFER_TOO_SMALL);
+    ReturnErrorCodeIf(!mFactoryData.vendor_name.data, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+    memcpy(buf, mFactoryData.vendor_name.data, mFactoryData.vendor_name.len);
+    ChipLogProgress(DeviceLayer, "vendor name: %s", buf);
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR FactoryDataProvider::GetVendorId(uint16_t & vendorId)
+{
+    ReturnErrorCodeIf(!mFactoryData.vendor_id.data, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+    memcpy(&vendorId, mFactoryData.vendor_id.data, mFactoryData.vendor_id.len);
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR FactoryDataProvider::GetProductName(char * buf, size_t bufSize)
+{
+    ReturnErrorCodeIf(bufSize < mFactoryData.product_name.len, CHIP_ERROR_BUFFER_TOO_SMALL);
+    ReturnErrorCodeIf(!mFactoryData.product_name.data, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+    memset(buf, 0, bufSize);
+    memcpy(buf, mFactoryData.product_name.data, mFactoryData.product_name.len);
```

Comment:
Again: does `mFactoryData.product_name.len` already include the null terminator?  Needs to be documented.

Reply:
This length does not include the null terminator. The factory data creation process takes into account how the product name needs to be formatted such that it is copied in properly to the buffer. More documentation on this process can be found in docs/guides/ti

Reply:
added a check for the buffer to have space for a null terminator, and added in the null terminator after the memcpy.